tests/components/scrape/test_init.py

```diff
@@ -49,14 +50,16 @@ async def test_setup_no_data_fails(hass: HomeAssistant) -> None:
     }
 
     with patch(
-        "homeassistant.components.scrape.coordinator.RestData",
+        "homeassistant.components.rest.RestData",
         return_value=MockRestData("test_scrape_sensor_no_data"),
     ):
         assert await async_setup_component(hass, DOMAIN, config)
         await hass.async_block_till_done()
 
     state = hass.states.get("sensor.ha_version")
-    assert state.state == STATE_UNAVAILABLE
+    assert state is None
+
+    assert "Platform scrape not ready yet" in caplog.text
```

Comment:
It's better to check that the platform is set up ok after the issue is resolved and time is moved forward.

Reply:
It has now been addressed so it can recover and also extended the test to ensure it does